prover/setup.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# This script sets up the necessary data needed by the CPU FRI prover to be used locally.
+
+if [[ -z "${ZKSYNC_HOME}" ]]; then
+  echo "Environment variable ZKSYNC_HOME is not set. Make sure it's set and pointing to the root of this repository"
+  exit 1
+fi
+
+sed -i.backup 's/^proof_sending_mode=.*$/proof_sending_mode="OnlyRealProofs"/' ../etc/env/base/eth_sender.toml
+sed -i.backup 's/^proof_loading_mode=.*$/proof_loading_mode="FriProofFromGcs"/' ../etc/env/base/eth_sender.toml
+rm ../etc/env/base/eth_sender.toml.backup
+sed -i.backup 's/^setup_data_path=.*$/setup_data_path="vk_setup_data_generator_server_fri\/data\/"/' ../etc/env/base/fri_prover.toml
+rm ../etc/env/base/fri_prover.toml.backup
+sed -i.backup 's/^universal_setup_path=.*$/universal_setup_path="..\/keys\/setup\/setup_2^26.key"/' ../etc/env/base/fri_proof_compressor.toml
+rm ../etc/env/base/fri_proof_compressor.toml.backup
+
+zk config compile dev
+
+for i in {1..13}
+do
+    zk f cargo run --release --bin zksync_setup_data_generator_fri -- --numeric-circuit $i --is_base_layer
```

Comment:
these take a lot of time -- I wonder if we should somehow detect that the files are already present?

Reply:
I added some basic detection for this in the setup script. https://github.com/matter-labs/zksync-era/pull/258/commits/3435d1b8ce22056cad473965b937e3e7f59a1c72